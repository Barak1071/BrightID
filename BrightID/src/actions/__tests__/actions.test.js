import * as action from '../index';

describe('actions', () => {
  test('should create an action to update the user trust score', () => {
    const score = '99.9';
    expect(action.setUserScore(score)).toMatchSnapshot();
  });

  test('should create an action to update the trust score of a connection', () => {
    const publicKey = [];
    const score = '88.6';
    expect(
      action.connectionScore(publicKey, score),
    ).toMatchSnapshot();
  });

  test('should create an action to set the group count', () => {
    const groupsCount = '15';
    expect(action.setGroupsCount(groupsCount)).toMatchSnapshot();
  });

  test('should create an action to set the search param', () => {
    const searchParam = 'gandolf';
    expect(action.setSearchParam(searchParam)).toMatchSnapshot();
  });

  test('should create an action to set the connections list', () => {
    const connections = [];
    expect(action.setConnections(connections)).toMatchSnapshot();
  });

  test('should create an action to add a connection', () => {
    const connection = {};
    expect(action.addConnection(connection)).toMatchSnapshot();
  });

  test('should create an action to set user data', () => {
    const userData = {
      publicKey: [],
      secretKey: [],
      name: 'aragon',
      photo: 'photo string',
    };
    expect(action.setUserData(userData)).toMatchSnapshot();
  });

  test('should create an action to remove user data', () => {
    expect(action.removeUserData()).toMatchSnapshot();
  });

  test('should create an action to set user photo', () => {
    const photo = 'photo string';
    expect(action.setPhoto(photo)).toMatchSnapshot();
  });

  test('should create an action to set public key 1', () => {
    const publicKey2 = [];
    expect(action.setPublicKey2(publicKey2)).toMatchSnapshot();
  });

  test('should create an action to set public key 1', () => {
    const nearbyPeople = [];
    expect(action.refreshNearbyPeople(nearbyPeople)).toMatchSnapshot();
  });

  test('should create an action to set public key 1', () => {
    const errmsg = 'error message';
    expect(action.handleError(errmsg)).toMatchSnapshot();
  });
});
