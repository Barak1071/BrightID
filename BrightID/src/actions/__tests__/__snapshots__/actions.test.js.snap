// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create an action to add a connection 1`] = `
Object {
  "connection": Object {},
  "type": "ADD_CONNECTION",
}
`;

exports[`actions should create an action to remove user data 1`] = `
Object {
  "type": "REMOVE_USER_DATA",
}
`;

exports[`actions should create an action to set public key 1 1`] = `
Object {
  "publicKey2": Array [],
  "type": "PUBLICKEY2",
}
`;

exports[`actions should create an action to set public key 1 2`] = `
Object {
  "nearbyPeople": Array [],
  "type": "REFRESH_NEARBY_PEOPLE",
}
`;

exports[`actions should create an action to set public key 1 3`] = `
Object {
  "error": "error message",
  "type": "ERROR",
}
`;

exports[`actions should create an action to set the connections list 1`] = `
Object {
  "connections": Array [],
  "type": "UPDATE_CONNECTIONS",
}
`;

exports[`actions should create an action to set the group count 1`] = `
Object {
  "groupsCount": "15",
  "type": "GROUPS_COUNT",
}
`;

exports[`actions should create an action to set the search param 1`] = `
Object {
  "type": "SEARCH_PARAM",
  "value": "gandolf",
}
`;

exports[`actions should create an action to set user photo 1`] = `
Object {
  "type": "USER_PHOTO",
  "photo": "photo string",
}
`;

exports[`actions should create an action to set user data 1`] = `
Object {
  "name": "aragon",
  "publicKey": Array [],
  "secretKey": Array [],
  "type": "UPDATE_USER_DATA",
  "photo": "photo string",
}
`;

exports[`actions should create an action to update the user trust score 1`] = `
Object {
  "score": "99.9",
  "type": "USER_SCORE",
}
`;
